### ISSUE TITLE
recording does not support alpha

### ISSUE THREAD

**Issue by asinghvi17** [JuliaPlots/Makie.jl#518]
All colors are cast to RGBX in `recordframe!`, when that ideally should not be the case.  We could also use an input pixfmt of `argb` or something instead of `yuv420`.  

Is the constraint of being RGB only a GLMakie thing, or something else?

**Comment by ffreyer** [comment]
I guess #4150 is related

**Comment by asinghvi17** [comment]
Yep pretty much, I guess it depends on whether the scenes background color is transparent or not though.
